lunatrace/bsl/frontend/src/pages/vulnerabilities/detail/DetailBody.tsx

```diff
@@ -87,6 +89,31 @@ export const VulnerabilityDetailBody: React.FunctionComponent<VulnerabilityDetai
                 ) : (
                   <span>No CVSS score</span>
                 )}
+                <hr />
+                <h4 className="">
+                  Your Projects Vulnerable: <span className="lighter">{vuln.findings.length}</span>{' '}
+                </h4>
+
+                {vuln.findings.map((f) => {
+                  const projectName = f.default_branch_build?.project?.name;
+                  const buildLink = `/project/${f.default_branch_build?.project_id}/build/${
+                    f.default_branch_build?.id as string
+                  }`;
+                  return (
+                    <div key={f.id as string}>
```

Comment:
I'm not a huge fan of this because it can create more runtime errors. I prefer the more verbose, explicit flow generally:
```tsx
(f) => {
  const findingId = f.id;
  if (!findingId) {
    console.error('Invalid ID attempted to render');
    return null;
  }
  const defaultBranchBuildId = f.default_branch_build?.id;
  const defaultBranchProjectId = f.default_branch_build?.project_id;
  if (!defaultBranchBuildId || !defaultBranchProjectId) {
    // log and error
  }
  // ... return your rendered components without any casting now.
}
```